packages/@azure/servicebus/data-plane/lib/receiver.ts

```diff
@@ -0,0 +1,242 @@
+ */
+export class Receiver {
+  /**
+   * @property {ClientEntityContext} _context Describes the amqp connection context for the QueueClient.
+   */
+  private _context: ClientEntityContext;
+  private _receiveMode: ReceiveMode;
+
+  constructor(context: ClientEntityContext, options?: MessageReceiverOptions) {
+    this._context = context;
+    if (!options) {
+      options = {};
+    }
+    this._receiveMode =
+      options.receiveMode === undefined ? ReceiveMode.peekLock : options.receiveMode;
+  }
+
+  /**
+   * Registers handlers to deal with the incoming stream of messages over an AMQP receiver link
+   * from a Queue/Subscription.
+   * To stop receiving messages, call `close()` on the Receiver or set the property
+   * `maxMessageWaitTimeoutInSeconds` in the options to provide a timeout.
+   *
+   * @param onMessage - Handler for processing each incoming message.
+   * @param onError - Handler for any error that occurs while receiving or processing messages.
+   * @param options - Options to control whether messages should be automatically completed and/or
+   * automatically have their locks renewed. You can also provide a timeout in seconds to denote the
+   * amount of time to wait for a new message before closing the receiver.
+   *
+   * @returns void
+   */
+  receive(onMessage: OnMessage, onError: OnError, options?: MessageHandlerOptions): void {
+    this.validateNewReceiveCall(ReceiverType.streaming);
+
+    if (!options) options = {};
+    const rcvOptions: ReceiveOptions = {
+      maxConcurrentCalls: 1,
+      receiveMode: this._receiveMode,
+      autoComplete: options.autoComplete,
+      maxAutoRenewDurationInSeconds: options.maxAutoRenewDurationInSeconds
+    };
+    const sReceiver = StreamingReceiver.create(this._context, rcvOptions);
+    this._context.streamingReceiver = sReceiver;
+    return sReceiver.receive(onMessage, onError);
+  }
+
+  /**
+   * Returns a batch of messages based on given count and timeout over an AMQP receiver link
+   * from a Queue/Subscription.
+   *
+   * @param maxMessageCount      The maximum number of messages to receive from Queue/Subscription.
+   * @param maxWaitTimeInSeconds The maximum wait time in seconds for which the Receiver
+   * should wait to receive the first message. If no message is received by this time,
+   * the returned promise gets resolved to an empty array.
+   * - **Default**: `60` seconds.
+   * @returns Promise<ServiceBusMessage[]> A promise that resolves with an array of Message objects.
+   */
+  async receiveBatch(
+    maxMessageCount: number,
+    maxWaitTimeInSeconds?: number
+  ): Promise<ServiceBusMessage[]> {
+    this.validateNewReceiveCall(ReceiverType.batching);
+
+    if (!this._context.batchingReceiver || !this._context.batchingReceiver.isOpen()) {
+      const options: ReceiveOptions = {
+        maxConcurrentCalls: 0,
+        receiveMode: this._receiveMode,
+        maxMessageWaitTimeoutInSeconds: 1
```

Comment:
Can maybe rename this to "perMessageWaitTimeoutInSeconds"
and first parameter to "idleTimeoutInSeconds"